notifications/signals.py

```diff
@@ -0,0 +1,29 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from .models import Notification, Type
+from companies.models import CompanyFollowers
+from users.models import User
```

Comment:
Avoid importing the `User` model directly unless needed for flexibility and Django best practice.
```
from django.contrib.auth import get_user_model
User = get_user_model()
```